File: pc_builder_service/internal/domain/models/gpu.go
package models

import (
	"time"

	"github.com/google/uuid"
)


type GPU struct {
	ID 					uuid.UUID 				`gorm:"type:uuid;primaryKey"`
	Brand 				string 					`gorm:"type:varchar(50);not null"`
	Family 				string 					`gorm:"type:varchar(10);not null"`
	Series 				string 					`gorm:"type:varchar(20);not null"`
	MemoryAmount 		int32 					`gorm:"type:integer;not null"`
	MemoryGen 			string 					`gorm:"type:varchar(10);not null"`
	Cores 				int32 					`gorm:"type:integer;not null"`
	PciExpress 			int32 					`gorm:"column:pci_express;type:integer;not null"`
	RecommendedPower 	int32 					`gorm:"type:integer;not null"`
	ReleaseDate 		time.Time 				`gorm:"type:timestamp;not null"`
	AvgPrice 			float32 				`gorm:"type:real;not null"`
	Img 				[]byte 					`gorm:"type:bytea"`
	CreatedAt 			time.Time 				`gorm:"type:timestamp;not null"`
	UpdatedAt 			time.Time 				`gorm:"type:timestamp"`
}

File: benchmark_service/src/main/java/com/buildpc/benchmark_service/entities/GPU.java
package com.buildpc.benchmark_service.entities;

import com.google.type.DateTime;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.sql.Date;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "gpus")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class GPU {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private UUID id;
	private String brand;
	private String family;
	private String series;
	private Integer memoryAmount;
	private String memoryGen;
	private Integer cores;
	private Integer pciExpress;
	private Integer recommendedPower;
	private LocalDateTime releaseDate;
	private Float avgPrice;
	private byte[] img;

	@CreatedDate
	private LocalDateTime createdAt;

	@LastModifiedDate
	private LocalDateTime updatedAt;
}
